snap/info_snap_yaml_test.go

```diff
@@ -788,3 +789,121 @@ slots:
 	c.Check(slot2.Apps, DeepEquals, map[string]*snap.AppInfo{
 		app1.Name: app1, app2.Name: app2})
 }
+
+// type and architectures
+
+func (s *YamlSuite) TestSnapYamlTypeDefault(c *C) {
+	y := []byte(`name: fatbinary
+version: 1.0
+`)
+	info, err := snap.InfoFromSnapYaml(y)
+	c.Assert(err, IsNil)
+	c.Assert(info.Type, Equals, snap.TypeApp)
+}
+
+func (s *YamlSuite) TestSnapYamlMultipleArchitecturesParsing(c *C) {
+	y := []byte(`name: fatbinary
+version: 1.0
+architectures: [i386, armhf]
+`)
+	info, err := snap.InfoFromSnapYaml(y)
+	c.Assert(err, IsNil)
+	c.Assert(info.Architectures, DeepEquals, []string{"i386", "armhf"})
+}
+
+func (s *YamlSuite) TestSnapYamlSingleArchitecturesParsing(c *C) {
+	y := []byte(`name: fatbinary
+version: 1.0
+architectures: [i386]
+`)
+	info, err := snap.InfoFromSnapYaml(y)
+	c.Assert(err, IsNil)
+	c.Assert(info.Architectures, DeepEquals, []string{"i386"})
+}
+
+func (s *YamlSuite) TestSnapYamlNoArchitecturesParsing(c *C) {
+	y := []byte(`name: fatbinary
+version: 1.0
+`)
+	info, err := snap.InfoFromSnapYaml(y)
+	c.Assert(err, IsNil)
+	c.Assert(info.Architectures, DeepEquals, []string{"all"})
+}
+
+func (s *YamlSuite) TestSnapYamlBadArchitectureParsing(c *C) {
+	data := []byte(`name: fatbinary
```

Comment:
Just a nitpick but tests above use ``y`` and this one uses ``data``. Being consistent would look nicer.